core/src/query.rs

```diff
@@ -128,13 +133,19 @@ impl MongoStatement for MongoQuery {
         }}];
 
         let opt = AggregateOptions::builder().comment_bson(Some(stmt_id));
+
+        let mut max_time = None;
         // If the query timeout is 0, it means "no timeout"
-        let options = if self.query_timeout.is_some_and(|timeout| timeout > 0) {
-            opt.max_time(Duration::from_millis(self.query_timeout.unwrap() as u64))
-                .build()
-        } else {
-            opt.build()
-        };
+        if self.query_timeout.is_some_and(|timeout| timeout > 0) {
+            max_time = Some(Duration::from_millis(self.query_timeout.unwrap() as u64))
+        }
+
+        let mut batch_size = None;
+        // 100 is an arbitrary value
+        if rowset_size > 100 {
+            batch_size = Some(rowset_size)
```

Comment:
We still need to figure out what value to use here. I'm just arbitrarily using 100 at the moment.